### PR TITLE
Various tx improvements

### PR DESCRIPTION
**Pull request by prestwich**
* Removed 255 limit to `tx_ins` and `tx_outs`
* Removed dust error
* Added support for non-compact VarInts (because real life is messy :frowning:)
* Fixed incorrect handling of VarInt(255)
* Updated tests
* Bumped patch version

### CODE REVIEW

**Comment by prestwich** [comment]
This patch successfully parses all txns in this block:
https://www.blockchain.com/btc/block/000000000000000000170b3da1e60f139c603e659b378753023c96c275169eea

File: riemann/tx/tx.py [hunk 1]
```diff
@@ -138,7 +137,7 @@ class VarInt(ByteData):
     '''
     NB: number must be integer
     '''
-    def __init__(self, number):
+    def __init__(self, number, l=None):
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Ambiguous variable. Sad.

File: riemann/tx/tx.py [hunk 2]
```diff
@@ -149,17 +148,20 @@ def __init__(self, number):
                              .format(number))
         if number <= 0xfc:
             pass  # No prefix
-        elif number <= 0xffff:
+        elif number <= 0xffff or l == 3:
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Amb var

File: riemann/tx/tx.py [hunk 3]
```diff
@@ -149,17 +148,20 @@ def __init__(self, number):
...
             self += bytes([0xfd])
-        elif number <= 0xffffffff:
+            l = 3
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Amb var

File: riemann/tx/tx.py [hunk 4]
```diff
@@ -149,17 +148,20 @@ def __init__(self, number):
...
+        elif number <= 0xffffffff or l == 5:
             self += bytes([0xfe])
-        elif number <= 0xffffffffffffffff:
+            l = 5
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Amb var

File: riemann/tx/tx.py [hunk 5]
```diff
@@ -149,17 +148,20 @@ def __init__(self, number):
...
+        elif number <= 0xffffffffffffffff or l == 9:
             self += bytes([0xff])
+            l = 9
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Amb var

File: riemann/tx/tx.py [hunk 6]
```diff
@@ -191,12 +193,14 @@ def from_bytes(VarInt, byte_string):
...
-        ret = VarInt(utils.le2i(num))
-
-        if non_compact:
+        if non_compact and 'overwinter' in riemann.get_current_network_name():
             raise ValueError('VarInt must be compact. Got: {}'
                              .format(byte_string.hex()))
 
+        ret = VarInt(
+            utils.le2i(num),
+            l=len(num) + 1 if non_compact else 0)
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Amb var

File: riemann/tx/tx.py [hunk 7]
```diff
@@ -588,7 +582,7 @@ def from_bytes(Tx, byte_string):
                 current += len(tx_witness)
                 tx_witnesses.append(tx_witness)
 
-        lock_time = byte_string[current:]
+        lock_time = byte_string[current:current+4]
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Add white space around +
`lock_time = byte_string[current:current + 4]`

File: riemann/tx/tx.py [hunk 8]
```diff
@@ -149,17 +148,20 @@ def __init__(self, number):
                              .format(number))
         if number <= 0xfc:
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Add comments to if-else block for clarity

File: riemann/tx/tx.py [hunk 9]
```diff
@@ -138,7 +137,7 @@ class VarInt(ByteData):
     '''
     NB: number must be integer
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Add formatted doc string.

File: riemann/tx/tx.py [hunk 10]
```diff
@@ -191,12 +193,14 @@ def from_bytes(VarInt, byte_string):
...
                              .format(byte_string.hex()))
 
-        ret = VarInt(utils.le2i(num))
-
-        if non_compact:
+        if non_compact and 'overwinter' in riemann.get_current_network_name():
             raise ValueError('VarInt must be compact. Got: {}'
                              .format(byte_string.hex()))
 
+        ret = VarInt(
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Did you intended to use `ret` again in this method? Return statement directly if not.

File: riemann/tx/tx.py [hunk 11]
```diff
@@ -345,13 +349,6 @@ def __init__(self, value, output_script):
         self.validate_bytes(value, 8)
         self.validate_bytes(output_script, None)
 
-        if (output_script != b''
```
Comment (rrybarczyk) [inline riemann/tx/tx.py]: Review dust rules. OP_RETURN is an edge case.

File: riemann/tests/tx/test_tx.py [hunk 12]
```diff
@@ -345,15 +345,6 @@ def test_copy(self):
         self.assertEqual(tx_out, tx_out_copy)  # They should be equal
         self.assertIsNot(tx_out, tx_out_copy)  # But not the same object
 
-    def test_dust_limit_error(self):
-        with self.assertRaises(ValueError) as context:
-            tx.TxOut(utils.i2le_padded(5, 8), self.output_script)
-
-        self.assertIn(
```
Comment (rrybarczyk) [inline riemann/tests/tx/test_tx.py]: Review dust rules. OP_RETURN is an edge case.
